src/components/OnThisPageNav.vue

```diff
@@ -0,0 +1,158 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+<template>
+  <div class="OnThisPageNav">
+    <slot
+      :active="currentPageSection"
+      :sections="onThisPageSections"
+      :checkIsActive="checkIsActive"
+      :scrollTo="scrollTo"
+      :getItemClasses="getItemClasses"
+    >
+      <ul class="items">
+        <li
+          v-for="item in onThisPageSections"
+          :key="item.anchor"
+          :class="getItemClasses(item)"
+        >
+          <router-link
+            :to="item.url"
+            class="base-link"
+            @click.native="scrollTo(item)"
+          >
+            {{ item.title }}
+          </router-link>
+        </li>
+      </ul>
+    </slot>
+  </div>
+</template>
+
+<script>
+import throttle from 'docc-render/utils/throttle';
+import { waitFrames } from 'docc-render/utils/loading';
+import ScrollToElement from 'docc-render/mixins/scrollToElement';
+import { buildUrl } from 'docc-render/utils/url-helper';
+
+export default {
+  name: 'OnThisPageNav',
+  mixins: [ScrollToElement],
+  inject: {
+    store: {
+      default() {
+        return {
+          state: { onThisPageSections: [], currentPageSection: {} },
+        };
+      },
+    },
+  },
+  computed: {
+    onThisPageSections: ({ store, $route }) => store.state.onThisPageSections.map(item => ({
+      ...item,
+      url: buildUrl(`#${item.anchor}`, $route.query),
+    })),
+    currentPageSection: ({ store }) => store.state.currentPageSection || {},
+  },
+  async mounted() {
+    window.addEventListener('scroll', this.onScroll, false);
+    this.$once('hook:beforeDestroy', () => {
+      window.removeEventListener('scroll', this.onScroll);
+    });
+    // make sure everything is ready
+    await waitFrames(8);
+    this.onScroll();
+  },
+  methods: {
+    onScroll: throttle(function onScroll() {
```

Comment:
It seems like because the code block renders slower than the rest of the page, `onScroll` is then triggered, even when I'm first landing on a page with a URL like: `blahblah#see-also`. If the See Also section is short, then we would highlight the previous section.
Is there a way to always highlight the section in the anchor when the page first loads?

Reply:
Think the jump is fixed, take a look.

The scroll detection should work on mounted, so we know which items is the current highlighted one.